DPGAnalysis/SiStripTools/plugins/APVCyclePhaseProducerFromL1ABC.cc

```diff
@@ -186,30 +189,59 @@ void APVCyclePhaseProducerFromL1ABC::produce(edm::Event& iEvent, const edm::Even
   if (iEvent.run() >= _firstgoodrun) {
     Handle<L1AcceptBunchCrossingCollection> pIn;
     iEvent.getByToken(_l1abccollectionToken, pIn);
-
+    Handle<TCDSRecord> tcds_pIn;
+    iEvent.getByToken(_tcdsRecordToken, tcds_pIn);
+    const auto& tcdsRecord = *tcds_pIn.product();
+    bool useTCDS(tcds_pIn.isValid());
```

Comment:
The `tcds_pIn.product()` still throws an exception when the `TCDSRecord` does not exist because it is called also when `useTCDS == false`.

Reply:
Fixed.